frontend2/src/components/elements/SelectMenu.tsx

```diff
@@ -30,22 +37,27 @@ function SelectMenu<T extends React.Key | null | undefined>({
     [options],
   );
   const invalid = errorMessage !== undefined;
+
+  let stateStyle = DEFAULT;
+  if (disabled) stateStyle = DISABLED;
+  else if (invalid) stateStyle = INVALID;
+  else if (value === undefined) stateStyle = PLACEHOLDER;
+
   return (
     <div className={`relative ${invalid ? "mb-2" : ""} ${className}`}>
-      <Listbox value={value} onChange={onChange}>
+      <Listbox value={value} onChange={onChange} disabled={disabled}>
         <div className="relative">
           {label !== undefined && (
             <Listbox.Label>
               <FormLabel label={label} required={required} />
             </Listbox.Label>
           )}
           <Listbox.Button
-            className={`relative h-9 w-full truncate rounded-md bg-white py-1.5
-            pl-3 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300
-            focus:outline-none focus:ring-1 focus:ring-cyan-600 ui-open:ring-cyan-600
-            sm:text-sm sm:leading-6 ${invalid ? "ring-red-500" : ""}`}
+            className={`relative h-9 w-full truncate rounded-md bg-white py-1.5 pl-3 pr-10
+            text-left shadow-sm ring-1 ring-inset focus:outline-none focus:ring-1
+            sm:text-sm sm:leading-6 ${stateStyle}`}
           >
-            <span className={`${value === undefined ? "text-gray-400" : ""}`}>
+            <span>
               {value === undefined ? placeholder : valueToLabel.get(value)}
             </span>
             <div
```

Comment:
the little chevron on the select menu breaks b/c the select menu is nested in a dialog
https://github.com/tailwindlabs/headlessui/issues/2524

I will push a fix for this (adding ` ui-not-open:rotate-0`) to the div

Reply:
Ohh yeah that explains a lot... I nuked the styling on `SelectMenu` like writ large so feel free to change/fix it however lol